R/generatePartialDependence.R

```diff
@@ -443,46 +445,67 @@ print.PartialDependenceData = function(x, ...) {
 #'   This feature must have been an element of the \code{features} argument to
 #'   \code{\link{generatePartialDependenceData}} and is only applicable when said argument had length
 #'   greater than 1.
+#'   The feature must be a factor or an integer.
 #'   If \code{\link{generatePartialDependenceData}} is called with the \code{interaction} argument \code{FALSE}
 #'   (the default) with argument \code{features} of length greater than one, then \code{facet} is ignored and
 #'   each feature is plotted in its own facet.
-#'   Note that if any of the elements of the \code{features} argument of \code{\link{generatePartialDependenceData}}
-#'   are factors, they will be coerced to numerics.
 #'   Default is \code{NULL}.
 #' @param p [\code{numeric(1)}]\cr
 #'   If \code{individual = TRUE} then \code{sample} allows the user to sample without replacement
 #'   from the output to make the display more readable. Each row is sampled with probability \code{p}.
 #'   Default is \code{1}.
+#' @param data [\code{data.frame}]\cr
+#'   Data points to plot. Usually the training data. For survival and binary classification tasks a rug plot
+#'   wherein ticks represent failures or instances of the positive class are shown. For regression tasks
+#'   points are shown. For multiclass classification tasks ticks are shown and colored according to their class.
+#'   Both the features and the target must be included.
+#'   Default is \code{NULL}.
 #' @template ret_gg2
 #' @export
-plotPartialDependence = function(obj, geom = "line", facet = NULL, p = 1) {
+plotPartialDependence = function(obj, geom = "line", facet = NULL, p = 1, data = NULL) {
   assertClass(obj, "PartialDependenceData")
-  if (length(obj$features) > 2L & geom != "tile" & obj$interaction)
-    stop("To plot more than 2 features geom must be 'tile'!")
   assertChoice(geom, c("tile", "line"))
-  if (geom == "tile" & !obj$interaction)
+  if (obj$interaction & length(obj$features) > 2L & !geom == "tile")
+    stop("Cannot plot more than 2 features together with line plots.")
+  if (geom == "tile") {
+    if (!(obj$task.desc$type %in% c("regr", "surv"))) {
+      if (length(obj$task.desc$class.levels) > 2L)
+        stop("Only visualization of binary classification works with tiling!")
+    }
+    feat_classes = sapply(obj$data, class)
+    if (!obj$interaction)
       stop("generatePartialDependenceData was called with interaction = FALSE!")
+  }
+
+  if (!is.null(data)) {
+    assertDataFrame(data, col.names = "unique", min.rows = 1L,
+                    min.cols = length(obj$features) + length(obj$td$target))
+    assertSubset(obj$features, colnames(data), empty.ok = FALSE)
+  }
 
   if (!is.null(facet)) {
     assertChoice(facet, obj$features)
     if (!length(obj$features) %in% 2:3)
-      stop("generatePartialDependence must be called with two or three features to use this argument!")
+      stop("generatePartialDependenceData must be called with two or three features to use this argument!")
```

Comment:
i'm confused. this error message is in plotPartialDependence if the obj argument, which has class PartialDependenceData but is generated by generatePartialDependenceData, wasn't called with arguments that comport with what is requested in plotPartialDependence. this change is to fix an error, which was referencing generatePartialDependence, which isn't a function at all.